### ISSUE TITLE
gRPC channel reported as not shut down properly

### ISSUE THREAD

**Issue by rmartine-ias** [pact-foundation/pact-plugins#14]
For the projects where we've implemented use of the gRPC plugin, we're seeing big red "SEVERE" warning messages that look like the following, whenever we run pact tests:

```java
Dec 05, 2022 5:42:15 PM io.grpc.internal.ManagedChannelOrphanWrapper$ManagedChannelReference cleanQueue
SEVERE: *~*~*~ Channel ManagedChannelImpl{logId=1, target=[::1]:60118} was not shutdown properly!!! ~*~*~*
    Make sure to call shutdown()/shutdownNow() and wait until awaitTermination() returns true.
java.lang.RuntimeException: ManagedChannel allocation site
        at io.grpc.internal.ManagedChannelOrphanWrapper$ManagedChannelReference.<init>(ManagedChannelOrphanWrapper.java:93)
        at io.grpc.internal.ManagedChannelOrphanWrapper.<init>(ManagedChannelOrphanWrapper.java:53)
        at io.grpc.internal.ManagedChannelOrphanWrapper.<init>(ManagedChannelOrphanWrapper.java:44)
        at io.grpc.internal.ManagedChannelImplBuilder.build(ManagedChannelImplBuilder.java:631)
        at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:297)
        at io.pact.plugins.jvm.core.DefaultPluginManager$tryInitPlugin$1.invoke(PluginManager.kt:776)
        at io.pact.plugins.jvm.core.Utils.handleWith(Util.kt:42)
        at io.pact.plugins.jvm.core.DefaultPluginManager.tryInitPlugin(PluginManager.kt:772)
        at io.pact.plugins.jvm.core.DefaultPluginManager.initialisePlugin(PluginManager.kt:755)
        at io.pact.plugins.jvm.core.DefaultPluginManager.loadPlugin(PluginManager.kt:378)
        at au.com.dius.pact.consumer.dsl.PactBuilder.usingPlugin(PactBuilder.kt:97)
        at au.com.dius.pact.consumer.dsl.PactBuilder.usingPlugin$default(PactBuilder.kt:94)
        at au.com.dius.pact.consumer.dsl.PactBuilder.usingPlugin(PactBuilder.kt)
        at io.pact.example.grpc.maven.PactConsumerTest.calculateRectangleArea(PactConsumerTest.java:46)
...
```

This does not, as far as I can tell, seem to impact the actual functioning of the plugin framework.

This can be reproduced by:

```shell
$ cd pact-plugins/examples/gRPC/area_calculator/consumer-maven
$ mvn clean verify
```

I spent a little time trying to track down the cause, but didn't come up with anything definitive in the time allotted. Here's what I did figure out:

* The error is printed in `void calculateRectangleArea` when the `ManagedChannel` is built
  * This is not an issue with that channel, but one allocated prior. See [quote](https://github.com/grpc/grpc-java/issues/9662#issuecomment-1299327192) from a maintainer of grpc-java:
    > If using a debugger, you may notice you get that message when creating a channel. We check for orphaned channels when new channels are created. So you'll see we display the message within the connect(), but the message is talking about a previous channel construction. 
* The error is talking about the channel created in [initPlugin](https://github.com/pact-foundation/pact-plugins/blob/ce83c3c156e27c2aae29f43596cd0638ecadbaeb/drivers/jvm/core/src/main/kotlin/io/pact/plugins/jvm/core/PluginManager.kt#L768) (which I would've realized sooner if I had actually read the stack trace ;p)
* `shutdown` [is actually called](https://github.com/pact-foundation/pact-plugins/blob/ce83c3c156e27c2aae29f43596cd0638ecadbaeb/drivers/jvm/core/src/main/kotlin/io/pact/plugins/jvm/core/PluginManager.kt#L263), but the error prints before the channel is shut down (which makes sense - I think the channel does want to stay alive until the plugin manager is done talking to the plugins.)

Thank you for your hard work!

**Comment by rholshausen** [comment]
I think this is due to it trying an IP6 connection first, and then failing back to an IP4 one. That trace is from the first channel not being shutdown (IP6 one), but it doesn't affect anything because it was never used.

**Comment by rholshausen** [comment]
Released JVM Plugin Driver 0.4.1
